shell/src/main/java/org/apache/accumulo/shell/commands/ConfigCommand.java

```diff
@@ -353,6 +344,31 @@ public int execute(final String fullCommand, final CommandLine cl, final Shell s
     return 0;
   }
 
+  private void validatePropertyType(String property, String value, String fullCommand,
+      String tableName) throws BadArgumentException {
```

Comment:
It looks like instead of `tableName` the parameter is more generic name (either table name of namespace name)

Reply:
I'm not sure how to determine the best fix for this since `namespace` and `table` don't have good property separation. 

Only `ValidTableProperties` are defined in `Property.java`. 
There is not a separate `ValidNamespaceProperties` hashset defined. 
https://github.com/apache/accumulo/blob/339672a6abb13762c68e46a2d8ba49f1d0b0150e/core/src/main/java/org/apache/accumulo/core/conf/Property.java#L1378-L1381

You can also see the interchangeability in the property set code for `namespace`. It's just checking if it's a valid table property vs anything `namespace` specific.
 
https://github.com/apache/accumulo/blob/339672a6abb13762c68e46a2d8ba49f1d0b0150e/shell/src/main/java/org/apache/accumulo/shell/commands/ConfigCommand.java#L105-L106 

So, should namespaces have separate properties that aren't table properties?  
if so, I can rework this and also add a `isValidNamespacePropertyKey` method to `Property.java`

Reply:
Properties are a hierarchy `default <- system <- namespace <- table` with the more specific overriding the more generic.  I'm not sure of the top of my head, but I expect that all table properties are valid namespace properties.  There may be a few specific system only props, but in general they should also be override-able  in a namespace or table.